MaxText/train.py

```diff
@@ -318,7 +318,7 @@ def main(argv: Sequence[str]) -> None:
   jax.config.update('jax_default_prng_impl', 'unsafe_rbg')
   os.environ["TF_CPP_MIN_LOG_LEVEL"] = "0"
   os.environ["LIBTPU_INIT_ARGS"] = os.environ.get("LIBTPU_INIT_ARGS","") + " --xla_tpu_spmd_rng_bit_generator_unsafe=true"
-  cc.initialize_cache(os.path.expanduser("~/jax_cache"))
+  cc.initialize_cache(os.environ.get("JAX_CACHE_DIR", os.path.expanduser("~/jax_cache")))
```

Comment:
Nice, but let's do this  via MaxText's config?

Reply:
We can move this lower and then write to say `config.jax_cache_dir`?

Reply:
Sure that sounds reasonable, will update.